src/utils/fs.js

```diff
@@ -2,38 +2,76 @@ import Promise from 'es6-promise/lib/es6-promise/promise.js';
 import * as fs from 'fs';
 import { dirname } from './path.js';
 
+function readdir ( dir ) {
+	return new Promise( ( fulfil, reject ) => {
+		fs.readdir( dir, ( err ) => {
+			if ( err ) {
+				reject( err );
+			} else {
+				fulfil();
+			}
+		});
+	});
+}
+
+function mkdir ( dir ) {
+	return new Promise( ( fulfil, reject ) => {
+		fs.mkdir( dir, ( err ) => {
+			if ( err ) {
+				reject( err );
+			} else {
+				fulfil();
+			}
+		});
+	});
+}
+
 function mkdirpath ( path ) {
 	const dir = dirname( path );
-	try {
-		fs.readdirSync( dir );
-	} catch ( err ) {
-		mkdirpath( dir );
-		fs.mkdirSync( dir );
-	}
+	return readdir(dir).catch(()=>{
+		return mkdirpath(dir).then(()=>{
+			return mkdir(dir);
+		});
+	});
 }
 
 export function isFile ( file ) {
-	try {
-		const stats = fs.statSync( file );
-		return stats.isFile();
-	} catch ( err ) {
-		return false;
-	}
+	return new Promise( ( fulfil, reject ) => {
+		fs.stat( file, ( err, stats ) => {
+			if ( err ) {
+				reject( err );
+			} else if (stats.isFile()) {
+				fulfil(file);
+			} else {
+				reject();
+			}
+		});
+	});
 }
 
 export function writeFile ( dest, data ) {
-	return new Promise( ( fulfil, reject ) => {
-		mkdirpath( dest );
+	return mkdirpath(dest).then(()=>
+		new Promise( ( fulfil, reject ) => {
```

Comment:
wait I am returning the promise, due to the lack of { braces }, I can switch it to use curly braces though